packages/core/src/utils.ts

```diff
@@ -1344,12 +1345,20 @@ export const getMaxGasLimit = (
   blockGasLimit: bigint,
   chainId: bigint
 ): bigint => {
+  // Rootstock has an exceptionally low block gas limit (6.8 million), so we need to set the max gas
+  // limit very high, or else large contracts won't be deployable with Sphinx.
+  if (
+    chainId === fetchChainIdForNetwork('rootstock') ||
+    chainId === fetchChainIdForNetwork('rootstock_testnet')
+  ) {
+    return (blockGasLimit * BigInt(99)) / BigInt(100)
+  }
   // On Scroll and Scroll Sepolia, set the max gas limit to be 80% of the block gas limit. We set a
   // higher value for these networks because their block gas limit is low (10 million), which means
   // a lower max gas limit could cause large contract deployments to be unexecutable. Transactions
   // that use ~8.5M gas were executed quickly on Scroll Sepolia, so an 80% limit shouldn't
   // meaningfully impact execution speed.
-  if (shouldUseHigherMaxGasLimit(chainId)) {
+  else if (shouldUseHigherMaxGasLimit(chainId)) {
     return (blockGasLimit * BigInt(8)) / BigInt(10)
   }
```

Comment:
I would prefer not to break the pattern of having all chain-specific settings defined in the `SPHINX_NETWORKS` array. Could you please refactor this so it doesn't rely on the if statement that checks for rootstock and rootstock testnet?

I recommend the following general approach:
1. Remove the `useHigherMaxGasLimit` field from the `SPHINX_NETWORKS` array.
2. Introduce a new field `maxBlockGasLimit` multiple, an object with two fields, numerator and denominator. 
3. In this function, use the values from the `maxBlockGasLimit` for the network.
4. Make sure Rootstock and Rootstock testnet have 99% multiples. For any networks where `useHigherMaxGasLimit` was true, we should have 80% multiples. 

This approach keeps all configurations inside the `SPHINX_NETWORKS` array and introduces a pattern covering all networks requiring a different maximum batch size.

Reply:
I think simpler approach is to just use the block gas limit to determine the max batch size. This way, we don't need to manually enter a field (or fields) in the `SPHINX_NETWORKS` array for each network.

We could change the `getMaxGasLimit` function to be something like:
- If blockGasLimit <= 7 million: use a 99% multiple
- Else if blockGasLimit < 15 million: use an 80% multiple
- Else: use a 50% multiple

It seems like we're essentially following this pattern already

Reply:
This pattern also handles the case where a network changes their block gas limit at some point in the future. (I haven't seen this happen before, but I think it's possible)

Reply:
That's acceptable to me, I just don't want to have cases where we have logic hanging around that runs based on the target network chain id.